inspirehep/modules/workflows/utils.py

```diff
@@ -177,3 +179,43 @@ def download_file_to_workflow(workflow, name, url):
             req.raw.decode_content = True
             workflow.files[name] = req.raw
             return workflow.files[name]
+
+
+def start_workflow(workflow, data=None, workflow_object=None):
+    """Start a workflow creating any needed models.
+
+    Due to the task `invenio_workflows.start` being a celery task itself, it
+    delays to it's own execution the creation of the
+    `invenio_workflows.Workflow` db entry, rendering the task unretriable if
+    for any reason it gets stuck before that step (for example, if rabbitmq
+    flushes the queue, or if a worker acks the task without actually doing
+    anything).
```

Comment:
Currenyly changing the task will not solve it, so we need a wrapper to that... That is the `start_workflow` function for now, but at some point we might want to have something provided by the workflows themselves instead. Changing the workflows to have one model instead of two is a major refactor (quite major).
The `start_async` it's thr same as `delay` but allows you to programmatically pass the args and kwargs (*kwargs does not work well on tasks). Starting it synchronously solves the model creation, but blocks until the worflow halts, so no good either.